### PR TITLE
Fixing the DSC resource definitions

### PR DESCRIPTION
**Pull request by vrdmr**
Both the resources are proxy resources and send all the Proxy Resource properties in the service. This change is to reflect the state of the service.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by lmazuel — APPROVED** [review]
Disregard this approval...

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json [hunk 1]
```diff
@@ -1564,7 +1569,11 @@
           "description": "Gets or sets the id of the resource."
         }
       },
-      "x-ms-azure-resource": true,
+      "allOf": [
```
Comment (lmazuel) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json]: If adding proxyResource, please remove "name" and "id" (duplicate)

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json [hunk 2]
```diff
@@ -1220,6 +1220,11 @@
           "description": "Gets or sets the properties of the Dsc Compilation job."
         }
       },
+      "allOf": [
```
Comment (lmazuel) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json]: If adding proxyResource, please remove "name" and "id" (duplicate)

**Review by lmazuel — CHANGES_REQUESTED** [review]
Duplicate properties

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json (hunk 1 above)
Comment (vrdmr) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json]: Thanks for catching this. I'll remove it.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json (hunk 2 above)
Comment (vrdmr) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json]: Resolved.

**Review by lmazuel — APPROVED** [review]

**Comment by lmazuel** [comment]
There is still linter issues, but I feel like the purpose of this specific PR is fine. Try to merge this as atomic improvement to the spec.

**Comment by vrdmr** [comment]
Thanks @lmazuel - We are in the process of fixing the Linter issues and would get them down to 0 soon.

**Pull request merged**